Lib/test/test_userstring.py

```diff
@@ -26,6 +26,7 @@ def checkequal(self, result, object, methodname, *args, **kwargs):
             result,
             realresult
         )
+        self.assertEqual(type(result), type(realresult))
```

Comment:
Take a look at the comment above.
```
# we don't fix the arguments, because UserString can't cope with it
```

Reply:
I guess I'm confused. Why do we have
```python
def rpartition(self, sep): return self.data.rpartition(sep)
```
but also
```python
def rstrip(self, chars=None):
        return self.__class__(self.data.rstrip(chars))
```
? It seems arbitrary which methods return UserStrings and which return regular strings. Is there some reasoning behind this?

Reply:
I think there may be no any reasoning. It may be just omission.

But the last word belongs to the maintainer of the module -- @rhettinger.